quinn-proto/src/cid_generator.rs

```diff
@@ -86,3 +90,93 @@ impl ConnectionIdGenerator for RandomConnectionIdGenerator {
         self.lifetime
     }
 }
+
+/// Generates 8-byte connection IDs that can be efficiently
+/// [`validate`](ConnectionIdGenerator::validate)d
+pub struct KeyedConnectionIdGenerator {
+    key: Arc<dyn crypto::HmacKey>,
+    lifetime: Option<Duration>,
+}
+
+impl KeyedConnectionIdGenerator {
+    /// Create a generator with a random key
+    #[cfg(feature = "ring")]
+    pub fn new() -> Self {
```

Comment:
Maybe just skip `new()` and move the implementation into `Default`?

Reply:
I like how `new` is much more discoverable in rustdocs, though I recognize that we're not really consistent about having it.